flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/utils/TableTestBase.scala

```diff
@@ -795,19 +797,20 @@ abstract class TableTestUtilBase(test: TableTestBase, isStreamingMode: Boolean)
       val plannerDirPath = clazz.getResource("/").getFile.replace("/target/test-classes/", "")
       new File(s"$plannerDirPath/src/test/resources$resourceTestFilePath")
     }
-    if (file.exists()) {
+    if (!file.exists() || "true".equalsIgnoreCase(System.getenv(PLAN_TEST_FORCE_OVERWRITE))) {
+      Files.deleteIfExists(file.toPath)
+      file.getParentFile.mkdirs()
+      assertTrue(file.createNewFile())
+      val prettyJson = TableTestUtil.getPrettyJson(jsonPlanWithoutFlinkVersion)
+      Files.write(Paths.get(file.toURI), prettyJson.getBytes)
```

Comment:
Use `Paths.get(file.toURI)` also for deletion? I think it has proven to work in all setups.

Reply:
+1